### PR TITLE
GHA - Adding names to jobs and workflows

### PR DESCRIPTION
**Pull request by GuyAv46**
**Describe the changes in the pull request**

Adding names to jobs and workflows, and some cleanup

**Mark if applicable**

- [ ] This PR introduces API changes
- [ ] This PR introduces serialization changes

### CODE REVIEW

File: .github/workflows/task-get-latest-tag.yml [hunk 1]
```diff
@@ -18,6 +18,7 @@ on:
 
 jobs:
   get-tag: # Following best practices: https://docs.github.com/en/rest/releases/releases?apiVersion=2022-11-28
+    name: Get Latest Release Tag for ${{ inputs.repo }} ${{ inputs.prefix }}
```
Comment (raz-mon) [inline .github/workflows/task-get-latest-tag.yml:21]: Where do we use\see this name?

**Comment by raz-mon** [comment]
😍

File: .github/workflows/event-merge-to-queue.yml [hunk 2]
```diff
@@ -1,4 +1,5 @@
 name: Merge a pull request
+run-name: Validate ${{ github.ref_name }}
```
[the remark this answers is not in the document]
Comment (GuyAv46) [reply .github/workflows/event-merge-to-queue.yml:2]: Yes. Seems to work

File: .github/workflows/task-get-latest-tag.yml (hunk 1 above)
Comment (GuyAv46) [reply .github/workflows/task-get-latest-tag.yml:21]: `event-merge-to-queue` and `event-pull_request`

**Review by raz-mon — APPROVED** [review]

**Pull request merged**